models/Asset/Image/Thumbnail/Processor.php

```diff
@@ -460,7 +462,11 @@ public static function process(Asset $asset, Config $config, $fileSystemPath = n
         }
 
         if ($statusCacheEnabled) {
-            $asset->getDao()->addToThumbnailCache($config->getName(), $filename);
+            if (isset($filesize) && isset($width) && isset($height)) {
+                $asset->getDao()->addToThumbnailCache($config->getName(), $filename, $filesize, $width, $height);
```

Comment:
PHPStan warns that you might not reach the same code route. Meaning, your code might run into the lock which gets released.

Reply:
But it should be still possible when the image got written ... maybe it requires some more ifs for Stan, but in case the thumb was written to the local filesystem, there should be also a width, height and filesize, no?

Reply:
I think we need to figure out which paths actually write the image and which don't. The null check is basically that, but direct instead of indirect.

If you have suggestions how to fix this exactly, I'll be glad to do it.

Reply:
In other words, we need to be able to guarantee the variables are available when storing since we'll make them mandatory and non-nullable.

Reply:
LGTM!